src/aspire/denoising/denoiser_cov2d.py

```diff
@@ -18,72 +20,115 @@ class DenoiserCov2D(Denoiser):
     """
     Define a derived class for denoising 2D images using Cov2D method
     """
-    def __init__(self, src, basis, var_noise, ctf_info):
+    def __init__(self, src, basis, var_noise, ctf_info, star_file):
         """
         Initialize an object for denoising 2D images using Cov2D method
 
         :param src: The source object of 2D images with metadata
         :param basis: The basis method to expand 2D images
         :param var_noise: The estimated variance of noise
         :param ctf_info: Whether the CTF filters are included
```

Comment:
Isn't this available from the `src` object?

Reply:
The src object only has the star file for input mrc files. This star file is for the output of denoised images.

Reply:
Same as above, the star file was removed from the DenoiseCov2D and used a new  `DenoisedImageSource` class for this in the new commit 23ca25c.